app/controllers/categories_controller.rb

```diff
@@ -306,6 +306,8 @@ def find
 
     if params[:ids].present?
       categories = Category.secured(guardian).where(id: params[:ids])
+      categories =
+        Category.secured(guardian).where(id: categories.select(:parent_category_id)) + categories
```

Comment:
I wonder is there a way to filter out NULL `parent_category_id` for `categories.select(:parent_category_id)` ?

Reply:
It would also be cool if we had a scope to get categories by ID as well as their parent categories...we come close to it with this scope but there is only a single category:

https://github.com/discourse/discourse/blob/1a1a63cf1ab4b81de1934c54f6664c7d46c9e87c/app/models/category.rb#L201-L212